io_ogre/ogre/node_anim.py

```diff
@@ -0,0 +1,122 @@
+        importlib.reload(util)
+
+# This is only relevant on first run, on later reloads those modules
+# are already in locals() and those statements do not do anything.
+import bpy, mathutils, logging, time
+from .. import config
+from ..report import Report
+from ..xml import RDocument
+from .. import util
+from os.path import join
+
+logger = logging.getLogger('node_anim')
+
+# Node Animation, based on the work done in Easy Ogre Exporter (3D Studio Max exporter)
+# https://github.com/OGRECave/EasyOgreExporter/blob/master/source/ExScene.cpp#L104
+# The idea is that the data exported by blender2ogre would have the same format as the one exported by Easy Ogre Exporter which would be the standard by being the first ones to implement it
+# There seems to be some support for animation in the Maya Ogre Exporter, but for the camera so it is possible to animate values like FOV
+# https://github.com/bitgate/maya-ogre3d-exporter/blob/master/src/ogreExporter.cpp#L373
+
+def dot_nodeanim(ob, doc, xmlnode):
+    """
+    Create the node animation for this object. 
+    This is only possible if the object has any animation data
+
+    ob: the blender object
+    doc: the parent xml node to attach the animation data
+    """
+
+    write_animations2(ob, doc, xmlnode)
+    
+# This function processess the objects animation frame by frame, insted of using its keyframes
+# The advantage of this method is that it respects the users  tuning of the node animation curves
+# The disadvantage is that it generates more data than method 1 and might clutter the .scene file which is not a binary file so it might take a lot longer to process when loading the .scene
+def write_animations2(ob, doc, xmlnode):
+
+    anim = ob.animation_data
+    
+    if anim is None or anim.action is None:
+        return
+
+    logger.info('* Generating node animation for: %s' % ob.name)
+    
+    start = time.time()
+    
+    _fps = float( bpy.context.scene.render.fps )
+
+    frame_start, frame_end = anim.action.frame_range
+
+    # Actually in Blender this does not make sense because there is only one possible animation per object, but lets maintain compatibility with Easy Ogre Exporter
+    aa = doc.createElement('animations')
+    xmlnode.appendChild(aa)
+    
+    a = doc.createElement('animation')
+    a.setAttribute("name", "%s" % anim.action.name)
+    a.setAttribute("enable", "false")
+    a.setAttribute("loop", "false")
+    a.setAttribute("interpolationMode", "linear")
+    a.setAttribute("rotationInterpolationMode", "linear")
+    a.setAttribute("length", '%6f' % ((frame_end) / _fps))
+    aa.appendChild(a)
+
+    logger.info (" - Action name: %s" % anim.action.name)
+    logger.info (" - Frame range: %s -> %s" % (frame_start, frame_end))
+    
+    frame_current = bpy.context.scene.frame_current
+    
+    initial_location = mathutils.Vector((0, 0, 0))
+    
+    for frame in range(int(frame_start), int(frame_end) + 1):
+        kf = doc.createElement('keyframe')
```

Comment:
probably this can be merged with `_ogre_node_helper`

Reply:
I'm not so sure now of merging with _ogre_node_helper, because in the case of keyframes translation != position and it is not the same calculation.

I was in the process of creating a function _ogre_transform_helper() and then realized that it needs to get the "position", rotation and scale and in the end the code duplication ends up not being so big after all.

It might be better to leave them separate, besides if someone decides to make changes there they might unintentionally break the Node Animations. Of course your idea is that if the change is an improvement then Node Animations benefit as well.

Reply:
yeah.. on a second thought, you are right.